auction-app-frontend/src/hooks/useCategoriesPaginated.js

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react";
```

Comment:
You can move this inside the service and handle this there. 
Benefit of moving it there is that we will have one file for handling all of our requests for specific use case, while with this approach we will end up with multiple custom hook files. 
One even more important thing is that we are using useEffect here and also we have in a component. Imagine case where this useEffect and one inside of the component have the same value in the dependency array, we will have 2 re-renders instead of one, which can affect performances.